### PR TITLE
chore: Add instructions regarding tool versions and env variables in Readme

### LINKED ISSUE #8622: [Bug]: Missing METAMASK_BUILD_TYPE; set to 'main' for a standard release, or 'flask' for a canary flask release. The default value is 'main'.
### Describe the bug

Missing METAMASK_BUILD_TYPE; set to 'main' for a standard release, or 'flask' for a canary flask release. The default value is 'main'.

### Expected behavior

_No response_

### Screenshots/Recordings

_No response_

### Steps to reproduce

yarn start:ios

### Error messages or log output

_No response_

### Version

7.13.0

### Build type

None

### Device

mac

### Operating system

iOS

### Additional context

_No response_

### Severity

_No response_

### PR DESCRIPTION
**Pull request by smilingkylan**
## **Description**

The MetaMask Mobile Platform team has identified improvements to the read me as a way improving the developer experience and onboarding.

## **Related issues**

Fixes: #8622  improves the effectiveness of the read me. 

## **Manual testing steps**

1. Read the repository Readme.md file

### Before:

https://github.com/MetaMask/metamask-mobile/assets/6249205/2aad15cf-a542-4514-b112-7f25800b1387

### After:

https://github.com/MetaMask/metamask-mobile/assets/6249205/42d0ca2c-5e0a-4000-855c-461e95bb9778

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by NicolasMassart — DISMISSED** [review]
Looks good to me.

**Comment by kylanhurt** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by Cal-L — DISMISSED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ To learn how to develop MetaMask-compatible applications, visit our [Developer D
 
 ### Environment setup
 
-Before running the app, make sure your development environment has all the required tools.
+Before running the app, make sure your development environment has all the required tools. Several of these tools (ie Node and Ruby) may require specific versions in order to succcessfully build the app.
```
Comment (Cal-L) [inline README.md:26]:
```suggestion
Before running the app, make sure your development environment has all the required tools. Several of these tools (ie Node and Ruby) may require specific versions in order to successfully build the app.
```

**Review by Cal-L — COMMENTED** [review]
Left a comment

File: README.md (hunk 1 above)
Comment (kylanhurt) [reply README.md:26]: @Cal-L Good catch!

**Review by Cal-L — APPROVED** [review]
LGTM

**Pull request merged**